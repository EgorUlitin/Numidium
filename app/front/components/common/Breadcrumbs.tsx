import * as React from 'react'

import { Breadcrumb } from 'antd'
import { css } from 'emotion'


interface Props {
    breadcrumbs?: string[]
}

export default (props: Props) =>
    <Breadcrumb className={s.breadcrumb}>
        {!!props.breadcrumbs && props.breadcrumbs.map((item, i) =>
            <Breadcrumb.Item key={i}>{item}</Breadcrumb.Item>
        )}
    </Breadcrumb>


const s = {
    breadcrumb: css`
        margin: -8px 0 8px 0;
    `,
}
