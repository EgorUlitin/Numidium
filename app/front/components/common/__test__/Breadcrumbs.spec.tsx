import * as React from 'react'
import * as renderer from 'react-test-renderer'

import Breadcrumbs from '../Breadcrumbs'

test('`Breadcrumbs` renders correctly with items', () => {
    const breadcrumbs = [ 'Home', 'Profile' ]

    const tree = renderer
        .create(<Breadcrumbs breadcrumbs={breadcrumbs} />)
        .toJSON()

    expect(tree).toMatchSnapshot()
})

test('`Breadcrumbs` renders correctly withpout items', () => {
    const tree = renderer
        .create(<Breadcrumbs />)
        .toJSON()

    expect(tree).toMatchSnapshot()
})
