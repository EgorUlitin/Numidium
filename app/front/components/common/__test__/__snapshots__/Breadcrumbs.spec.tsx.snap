// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`breadcrumbs renders correctly with items 1`] = `
<div
  className="css-6kt5zf ant-breadcrumb"
  style={undefined}
>
  <span>
    <span
      className="ant-breadcrumb-link"
    >
      Home
    </span>
    <span
      className="ant-breadcrumb-separator"
    >
      /
    </span>
  </span>
  <span>
    <span
      className="ant-breadcrumb-link"
    >
      Profile
    </span>
    <span
      className="ant-breadcrumb-separator"
    >
      /
    </span>
  </span>
</div>
`;

exports[`breadcrumbs renders correctly withpout items 1`] = `
<div
  className="css-6kt5zf ant-breadcrumb"
  style={undefined}
/>
`;
